packages/shorebird_cli/test/src/executables/gradlew_test.dart

```diff
@@ -329,5 +329,110 @@ Make sure you have run "flutter build apk" at least once.''',
         },
       );
     });
+
+    group('exists', () {
+      late Directory tempDir;
+      setUp(() {
+        tempDir = setUpAppTempDir();
+      });
+
+      group('when gradlew does not exist', () {
+        test('returns false', () {
+          expect(gradlew.exists(tempDir.path), isFalse);
+        });
+      });
+
+      group('when gradlew exists', () {
+        group(
+          'when on unix based OSs',
+          () {
+            setUp(() {
+              File(
+                p.join(tempDir.path, 'android', 'gradlew'),
+              ).createSync(recursive: true);
+            });
+
+            test(
+              'returns true',
+              () {
+                expect(gradlew.exists(tempDir.path), isTrue);
+              },
+              testOn: 'linux || mac-os',
+            );
+          },
+        );
+
+        group(
+          'when on windows',
+          () {
+            setUp(() {
+              File(
+                p.join(tempDir.path, 'android', 'gradlew.bat'),
+              ).createSync(recursive: true);
+            });
+
+            test(
+              'returns true',
+              () {
+                expect(gradlew.exists(tempDir.path), isTrue);
+              },
+              testOn: 'windows',
+            );
+          },
+        );
+      });
+    });
+
+    group('version', () {
+      late Directory tempDir;
+
+      setUp(() {
+        when(() => platform.isLinux).thenReturn(true);
+        when(() => platform.isMacOS).thenReturn(false);
+        when(() => platform.isWindows).thenReturn(false);
```

Comment:
Same here re: testOn

Reply:
This one I am not sure if it is worth to run in all platforms, since all is being tested is the parsing of the output, which shouldn't change (at least the relevant part for us), no matter which platform we are.

And if I change to testOn, then I will need to write one test for each platform (or at least of for unix based and one for windows), because we need to mock the executable (which you can see a couple of lines above) path differently.

Lmk if this makes sense

Reply:
You don't need to write one for each platform. I'm just suggesting that we use a consistent method of targeting a platform for a test.

Reply:
right, changed to use `testOn` on all tests.